File: src/main/java/in/dragonbra/javasteam/steam/handlers/steamauthentication/AuthenticationException.java
package in.dragonbra.javasteam.steam.handlers.steamauthentication;

import in.dragonbra.javasteam.enums.EResult;

/**
 * Thrown when {@link SteamAuthentication} fails to authenticate.
 */
public class AuthenticationException extends Exception {

    private EResult result;

    /**
     * Initializes a new instance of the {@link AuthenticationException} class.
     *
     * @param message The message that describes the error.
     * @param result  The result code that describes the error.
     */
    public AuthenticationException(String message, EResult result) {
        super(message + " with result " + result);

        this.result = result;
    }

    /**
     * @return the result of the authentication request.
     */
    public EResult getResult() {
        return result;
    }
}

File: src/main/java/in/dragonbra/javasteam/steam/authentication/AuthenticationException.kt
package `in`.dragonbra.javasteam.steam.authentication

import `in`.dragonbra.javasteam.enums.EResult

/**
 * Thrown when [SteamAuthentication] fails to authenticate.
 */
@Suppress("unused")
class AuthenticationException : Exception {

    /**
     *  The result of the authentication request.
     */
    var result: EResult? = null
        private set

    /**
     * Initializes a new instance of the [AuthenticationException] class.
     */
    constructor() : super()

    /**
     * Initializes a new instance of the [AuthenticationException] class.
     *
     * @param message The message that describes the error.
     * @param result  The result code that describes the error.
     */
    constructor(message: String, result: EResult) : super("$message with result $result") {
        this.result = result
    }
}